diff --git a/compiler-rt/cmake/Modules/CompilerRTMockLLVMCMakeConfig.cmake b/compiler-rt/cmake/Modules/CompilerRTMockLLVMCMakeConfig.cmake
index 42ede171e41e..7dc3eca9c432 100644
--- a/compiler-rt/cmake/Modules/CompilerRTMockLLVMCMakeConfig.cmake
+++ b/compiler-rt/cmake/Modules/CompilerRTMockLLVMCMakeConfig.cmake
@@ -51,6 +51,8 @@ function(compiler_rt_mock_llvm_cmake_config_set_target_triple)
         RESULT_VARIABLE HAD_ERROR
         OUTPUT_VARIABLE COMPILER_OUTPUT
         OUTPUT_STRIP_TRAILING_WHITESPACE)
+    elseif("${CMAKE_C_COMPILER_ID}" STREQUAL "MSVC")
+      set(COMPILER_OUTPUT "x86_64-pc-windows-msvc")
     else()
       message(FATAL_ERROR
         "Fetching target triple from compiler \"${CMAKE_C_COMPILER_ID}\" "
